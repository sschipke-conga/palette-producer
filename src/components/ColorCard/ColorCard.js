import './ColorCard.scss';
import React from 'react';
import { ChromePicker } from 'react-color';
import PropTypes from 'prop-types'
import { IoIosLock, IoIosUnlock } from 'react-icons/io';

export const ColorCard = ({ index, color, isLocked, changeColor, toggleLock }) => {

  const colorStyles = {
    background: color,
  }

  return (
    <div className='ColorCard'>
      <div className='color' style={colorStyles}>
        {isLocked ?
          <IoIosLock className='lock' onClick={() => toggleLock(index, color)} /> :
          <IoIosUnlock className='lock' onClick={() => toggleLock(index, color)} />}
        <p className='hex'>{color}</p>
      </div>
      <div className='picker-container'>
        <ChromePicker className='color-picker' disableAlpha={true} width={170} color={color} onChange={({ hex }) => changeColor(index, hex)} />
      </div>
    </div >
  )
}

export default ColorCard;

ColorCard.propTypes = {

}
