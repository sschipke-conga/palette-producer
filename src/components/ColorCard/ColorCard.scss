.ColorCard {
  display: flex;
  flex-direction: column;
  width: 20%;
}

.color {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 30vh;
  padding: 20px;
}

.lock {
  cursor: pointer;
  background-color: rgba(244, 244, 244, 0.3);
  border-radius: 5px;
  height: 50px;
  margin: 0 auto;
  width: 50px;
}

.hex {
  bottom: 0px;
  color: whitesmoke;
  font-size: 20px;
  font-weight: 500;
  left: 50%; 
  position: absolute;
  text-shadow: 0.5px 0.5px rgb(113, 113, 113);
  text-transform: uppercase;
  transform: translate(-50%);
}

.picker-container {
  display: flex;
  justify-content: center;
  margin-top: 5px;
  width: 100%;
  min-height: 12.5em;
}

.color-picker {
  width: 90%
}