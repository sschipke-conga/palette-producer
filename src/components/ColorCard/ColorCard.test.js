import React from "react";
import ColorCard from "./ColorCard";
import { shallow } from "enzyme";

describe('ColorCard', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ColorCard 
    index={3}
    color={'#FFFFFF'}
    isLocked={true}
    changeColor={jest.fn()}
    toggleLock={jest.fn()}
    />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
