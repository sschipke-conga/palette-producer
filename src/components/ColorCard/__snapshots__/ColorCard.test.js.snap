// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCard alt snapshot should match the alt snapshot if it is not locked 1`] = `undefined`;

exports[`ColorCard should match the snapshot 1`] = `undefined`;
